properly support job host summary data in custom notification templates (and remove a few fields that don't work)

ryanpetrello opened the pull request:
see: https://github.com/ansible/tower/issues/4148

ryanpetrello committed:
properly support job host summary data in custom notification templates

see: https://github.com/ansible/tower/issues/4148

ryanpetrello commented:
![image](https://user-images.githubusercontent.com/214912/75812007-74ba6080-5d5b-11ea-84aa-91107dfb52be.png)

![image](https://user-images.githubusercontent.com/214912/75812034-7d129b80-5d5b-11ea-97ad-b72a40cab386.png)

gamuniz commented:
you think you could extend this to also include the other options that aren't being populated:

{'playbook_counts': ['play_count', 'task_count']},

{'project_update': ['id', 'name', 'description', 'status', 'failed']}

{'source_workflow_job': ['description', 'elapsed', 'failed', 'id', 'name', 'status']}

ryanpetrello committed:
remove a few custom notification fields that don't work

these aren't top-level serializer fields; they're summary fields
if we want to support these properly, we should treat them as
enhancements, and write support, tests, and documentation

ryanpetrello commented:
@gamuniz I've removed those, because they don't work.

@jladdjr what's the process for getting @tvo318 what she needs to correct these fields I removed?

ryanpetrello commented on awx/main/models/notifications.py:
```diff
@@ -270,21 +270,19 @@ class JobNotificationMixin(object):
                             'scm_revision', 'diff_mode', 'job_slice_number', 'job_slice_count', 'custom_virtualenv',
                             'approval_status', 'approval_node_name', 'workflow_url',
-                            {'host_status_counts': ['skipped', 'ok', 'changed', 'failures', 'dark']},
-                            {'playbook_counts': ['play_count', 'task_count']},
```
This doesn't work (and from what I can tell, never did).

ryanpetrello commented on awx/main/models/notifications.py:
```diff
@@ -270,21 +270,19 @@ class JobNotificationMixin(object):
                                                                'total_inventory_sources', 'inventory_sources_with_failures',
                                                                'organization_id', 'kind']},
                                                 {'project': ['id', 'name', 'description', 'status', 'scm_type']},
-                                                {'project_update': ['id', 'name', 'description', 'status', 'failed']},
```
This doesn't work (and from what I can tell, never did).

ryanpetrello commented on awx/main/models/notifications.py:
```diff
@@ -270,21 +270,19 @@ class JobNotificationMixin(object):
                                                 {'instance_group': ['name', 'id']},
                                                 {'created_by': ['id', 'username', 'first_name', 'last_name']},
-                                                {'labels': ['count', 'results']},
-                                                {'source_workflow_job': ['description', 'elapsed', 'failed', 'id', 'name', 'status']}]}]
```
This doesn't work (and from what I can tell, never did).

jladdjr commented:
> @jladdjr what's the process for getting @tvo318 what she needs to correct these fields I removed?

@ryanpetrello - I think the two relevant docs that would need to be updated are:

* https://github.com/ansible/product-docs/blob/master/userguide/source/notifications.rst#create-custom-notifications
* https://github.com/ansible/product-docs/blob/master/installandreference/source/notification_parameters_supported.rst

The latter page has a detailed break-down of the fields that can be used in custom notifications.

ryanpetrello commented:
https://github.com/ansible/product-docs/pull/1021
